### PR TITLE
Fixed non-decreasing count of stream on H3 connection.

### LINKED ISSUE #49617: HTTP/3: Hang after 100 requests on a connection
HttpClient + HTTP/3 appears to consistently hang when making the 101st HTTP/3 requests to the server. I've looked at client and server logs and I believe the issue is HttpClient never sends the 101st request. The app calls `HttpClient.SendAsync`, nothing is received by the server, and the app hangs.

I'm guessing there is a setting that limits the number of concurrent streams to the server and that number isn't being reset by the client when requests are complete. When the 101st request is made the client thinks it has 100 requests in progress and hangs forever.

### PR DESCRIPTION
**Pull request by ManickaP**
Note that this fixes the problem only for the initially set limit of 100 streams. The more general fix and API will follow with https://github.com/dotnet/runtime/issues/32079. This is mainly to unblock ASP.NET core scenarios.

Fixes #49617

cc: @JamesNK

### CODE REVIEW

File: src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3Connection.cs [hunk 1]
```diff
@@ -295,15 +295,22 @@ private void OnMaximumStreamCountIncrease(long newMaximumStreamCount)
                     return;
                 }
 
-                _requestStreamsRemaining += (newMaximumStreamCount - _maximumRequestStreams);
+                IncreaseRemainingStreamCount(newMaximumStreamCount - _maximumRequestStreams);
                 _maximumRequestStreams = newMaximumStreamCount;
+            }
+        }
+
+        private void IncreaseRemainingStreamCount(long delta)
```
Comment (wfurt) [inline src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3Connection.cs:303]: if `delta` is negative "AdjustRemainingStreamCount" may be better name.

**Review by wfurt — APPROVED** [review]
LGTM

**Review by JamesNK — APPROVED** [review]
I'm not familiar with client code. Approving to show my enthusiasm to have this fixed 😄 

Once there is a daily SDK with this fix available then I'll test it out.

File: src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3Connection.cs (hunk 1 above)
Comment (ManickaP) [reply src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3Connection.cs:303]: It's never `<= 0`. I added an assert to express that.

**Pull request merged**